File: _ejemplos/pi/pi.c
#include <stdio.h>
#include <stdlib.h>

#define N 10000000

int main(){
   
   double step,x,pi,sum=0.0;

   step=1.0/N;
   for (int i=0;i<N;i++){
   	x=(i+0.5)*step;
   	sum+=4.0/(1.0+x*x);
   }
   pi=step*sum;
   printf("PI=%12.10f\n",pi);

   return 0;
}

File: _ejemplos/pi/pi.f90
program integral
 !Calcular la integral de f(x)=4/(1+x*x) entre 0 y 1 (es == PI)
 implicit none
 integer, parameter :: N=10000000
 double precision :: step,x,pi,sum=0.0
 integer :: i

  step=1.0/N
  do i=1,N
    x=(i+0.5)*step
    sum=sum+4.0/(1.0+x*x)
  enddo
  pi=step*sum
  print '("PI= ",f12.10)',pi
end program
